packages/table/preview/index.tsx

```diff
@@ -531,3 +531,119 @@ ReactDOM.render(
+    data?: any[];
+}
+
+const REORDERABLE_TABLE_DATA = [
+    ["A", "Apple", "Ape", "Albania", "Anchorage"],
+    ["B", "Banana", "Boa", "Brazil", "Boston"],
+    ["C", "Cranberry", "Cougar", "Croatia", "Chicago"],
+    ["D", "Dragonfruit", "Deer", "Denmark", "Denver"],
+    ["E", "Eggplant", "Elk", "Eritrea", "El Paso"],
+].map((array) => ({
+    // tslint:disable:object-literal-sort-keys
+    letter: array[0],
+    fruit: array[1],
+    animal: array[2],
+    country: array[3],
+    city: array[4],
+    // tslint:enable:object-literal-sort-keys
+}));
+
+class ReorderableTableExample extends React.Component<{}, IReorderableTableExampleState> {
+    public state = {
+        data: REORDERABLE_TABLE_DATA,
+    } as IReorderableTableExampleState;
+
+    public componentDidMount() {
+        const children = [
+            <Column key="1" name="Letter" renderCell={this.renderLetterCell} />,
+            <Column key="2" name="Fruit" renderCell={this.renderFruitCell} />,
+            <Column key="3" name="Animal" renderCell={this.renderAnimalCell} />,
+            <Column key="4" name="Country" renderCell={this.renderCountryCell} />,
+            <Column key="5" name="City" renderCell={this.renderCityCell} />,
+        ];
+        this.setState({ children });
+    }
+
+    public render() {
+        return (
+            <Table
+                isColumnReorderable={true}
+                isRowReorderable={true}
+                numRows={this.state.data.length}
+                onColumnsReordered={this.handleColumnsReordered}
+                onRowsReordered={this.handleRowsReordered}
+            >
+                {this.state.children}
+            </Table>
+        );
+    }
+
+    private renderLetterCell = (row: number) => <Cell>{this.state.data[row].letter}</Cell>;
+    private renderFruitCell = (row: number) => <Cell>{this.state.data[row].fruit}</Cell>;
+    private renderAnimalCell = (row: number) => <Cell>{this.state.data[row].animal}</Cell>;
+    private renderCountryCell = (row: number) => <Cell>{this.state.data[row].country}</Cell>;
+    private renderCityCell = (row: number) => <Cell>{this.state.data[row].city}</Cell>;
+
+    private handleColumnsReordered = (oldIndex: number, newIndex: number, length: number) => {
+        if (oldIndex === newIndex) { return; }
+        const nextChildren = this.reorderElementsInArray(this.state.children, oldIndex, newIndex, length);
+        this.setState({ children: nextChildren });
+    }
+
+    private handleRowsReordered = (oldIndex: number, newIndex: number, length: number) => {
+        if (oldIndex === newIndex) { return; }
+        this.setState({ data: this.reorderElementsInArray(this.state.data, oldIndex, newIndex, length) });
+    }
+
+    private reorderElementsInArray(array: any[], from: number, to: number, length: number) {
+        const before = array.slice(0, from);
+        const within = array.slice(from, from + length);
+        const after = array.slice(from + length);
```

Comment:
Oh i see. Then iterating is fine. And extracting this to a util method would defo make sense